include/powerloader/powerloader.hpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2021, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#ifndef POWERLOADER_API_HPP
+#define POWERLOADER_API_HPP
+
+/*
+// clang-format off
+#ifdef POWERLOADER_STATIC
+// As a static library: no symbol import/export.
+#  define POWERLOADER_API
+#else
+// As a shared library: export symbols on build, import symbols on use.
+#  ifdef POWERLOADER_EXPORTS
+     // We are building this library
+#    ifdef _MSC_VER
+#         define POWERLOADER_API __declspec(dllexport)
+#    else
+#         define POWERLOADER_API __attribute__((__visibility__("default")))
+#    endif
+#  else
+     // We are using this library
+#    ifdef _MSC_VER
+#         define POWERLOADER_API __declspec(dllimport)
+#    else
+#         define POWERLOADER_API // Symbol import is implicit on non-msvc compilers.
+#    endif
+#  endif
+#endif
+// clang-format on*/
+
+// Project version
```

Comment:
I would prefer version information to be in a separate (maybe included here) `version.hpp` headers, so that it can potentially be generated automatically later.

Reply:
We tend to do the opposite, we change the sourcd file, and read it from where we need. Thi way, the source does not depend on external tools.

Reply:
Ah ok then. I'm used to `version.hpp` having only the version info but it you thin it's not necessary to have it separate then ok.